collections/gamesCollection.mjs

```diff
@@ -45,21 +47,23 @@ class GamesCollection {
 
     url += "?format=json";
 
-    this.yf.api(this.yf.GET, url, (e, data) => {
-      if (e) {
-        return cb(e);
-      }
-
-      const games = parseCollection(data.fantasy_content.games, subresources);
-      return cb(null, games);
-    });
+    return this.yf.api(this.yf.GET, url)
+      .then(data => parseCollection(data.fantasy_content.games, subresources))
+      .then(games => { 
+        cb(null, games); 
+        return games; 
+      })
+      .catch(e => { 
```

Comment:
The general pattern for most invocations of yf.api was to take the "raw data => mapped object" portion of the callback and run that after yf.api returns. Then always invoke the callback when the promise resolves with the data, and invoke it with the error when it rejects.

I also considered adding a mappingfunc parameter to yf.api a and letting yf.api do all the callback invoking. To do that and still differentiate between the mapping function and a callback, It would probably require a yf.api method that accepted named parameters (i.e. an object). It would result in less changes (and total code) in each of these resources, but I wory it is further from the current API.

I might open a separate branch with that pattern, just to see what it looks like.